python/GafferUI/ColorChooser.py

```diff
@@ -389,49 +405,174 @@ def __positionToColor( self, position ) :
+				self.__DragConstraints.X in self.__dragConstraints or not self.__DragConstraints.Y in self.__dragConstraints
+			) :
+				# Constrain to horizontal line
+				return imath.V2f( position.x, self.__constraintPosition.y )
+			# Constrain to vertical line
+			return imath.V2f( self.__constraintPosition.x, position.y )
+
+		center = imath.V2f( self.bound().size() ) * 0.5
+		distanceToCenter = ( position - center ).length()
+
+		polarConstraintPosition = self.__cartesianToPolar( self.__constraintPosition, center, center.x )
+
+		# Extend line segment past the widget for distance check
+		extended = self.__polarToCartesian( imath.V2f( polarConstraintPosition.x, 2.0 ), center, center.x )
+
+		radiusLine = imath.Line3f( imath.V3f( center.x, center.y, 0 ), imath.V3f( extended.x, extended.y, 0 ) )
+		position3 = imath.V3f( position.x, position.y, 0 )
+		closestPointOnRadiusLine = radiusLine.closestPointTo( position3 )
+		distanceToRadiusLine = ( closestPointOnRadiusLine - position3 ).length()
+
+		constraintRadius = polarConstraintPosition.y * center.x
+		distanceToCircle = abs( distanceToCenter - constraintRadius )
+
+		dot = ( position - center ).dot( self.__constraintPosition - center )
+
+		polarPosition = self.__cartesianToPolar( position, center, center.x )
+
+		if self.__DragConstraints.X in self.__dragConstraints :
+			if (
+				abs( distanceToCircle ) < distanceToRadiusLine or
+				( dot < 0 and distanceToCenter > constraintRadius * 0.5 )
+			) :
+				# If we're closer to the circle than the radius line, or we're behind the center
+				# (from the point of view of `__constraintPosition`) and reasonably far from the center,
+				# constrain to the circle.
+				return self.__polarToCartesian(
+					imath.V2f( polarPosition.x, polarConstraintPosition.y ), center, center.x
+				)
+
+		if dot < 0 :
+			# If we're not constrained to the circle, but "behind" the center, snap to the center.
+			return center
+
+		# If no constraints are enforced yet, constrain to the radius line.
+		polarClosestPoint = self.__cartesianToPolar(
+			imath.V2f( closestPointOnRadiusLine.x, closestPointOnRadiusLine.y ), center, center.x
+		)
+		return self.__polarToCartesian(
+			imath.V2f( polarConstraintPosition.x, polarClosestPoint.y ),
+			center,
+			center.x
+		)
+
+	def __setupConstraints( self, event ) :
+
+		c, staticComponent = self.getColor()
+
+		self.__constraintPosition = self.__colorToPosition( c )
+
+		center = imath.V2f( self.bound().size() ) * 0.5
+
+		# Set `__dragConstraints` with `__constraintPosition` based on actual color
+		self.__dragConstraints = None
+		if event.modifiers == GafferUI.ButtonEvent.Modifiers.Control :
+			if not self.__useWheel() :
+				self.__dragConstraints = self.__DragConstraints.X | self.__DragConstraints.Y
+			else :
+				self.__dragConstraints = self.__DragConstraints.Y
+
+				center = imath.V2f( self.bound().size() ) * 0.5
+				if ( self.__constraintPosition - center ).length() > self.__snapRadius :
```

Comment:
Is `self.__snapRadius` still a useful name in light of the recent snap removal? It seems now only used here to specify the minimum radius we allow constraining to the circle. 

The increased value of 10 also still seems too high to me. On smaller wheels it limits the range of saturations and values that are hue-adjustable quite severely, down to not letting you make a constrained hue adjustment below a saturation or value of 0.3 on our smallest wheel. Maybe we can lower this to 4? That allows hue adjustments of a greater range of lower saturation colours while still not being _too_ twitchy.

Reply:
That's a good point, I renamed `__snapRadius` to `__minDraggableHueRadius` in https://github.com/GafferHQ/gaffer/pull/6128/commits/969217d4d97d05b76dd415e9f5ad813146eac790. I also changed the value to 4.0 as you recommend. I can now get the hue circle at saturations as low as 0.09 with the 3 slider wheel size.